crates/astria-sequencer/src/bridge/bridge_unlock_action.rs

```diff
@@ -0,0 +1,198 @@
+use anyhow::{
+    ensure,
+    Context as _,
+    Result,
+};
+use astria_core::{
+    primitive::v1::Address,
+    protocol::transaction::v1alpha1::action::{
+        BridgeUnlockAction,
+        TransferAction,
+    },
+};
+use tracing::instrument;
+
+use crate::{
+    accounts::action::transfer_check_stateful,
+    bridge::state_ext::StateReadExt as _,
+    state_ext::{
+        StateReadExt,
+        StateWriteExt,
+    },
+    transaction::action_handler::ActionHandler,
+};
+
+#[async_trait::async_trait]
+impl ActionHandler for BridgeUnlockAction {
+    async fn check_stateful<S: StateReadExt + 'static>(
+        &self,
+        state: &S,
+        from: Address,
+    ) -> Result<()> {
+        let transfer_action = TransferAction {
+            to: self.to,
+            asset_id: self.asset_id,
+            amount: self.amount,
+            fee_asset_id: self.fee_asset_id,
+        };
+
+        // ensure the from is a bridge account.
+        state
+            .get_bridge_account_rollup_id(&from)
+            .await
+            .context("failed to get bridge account rollup id")?
+            .ok_or_else(|| anyhow::anyhow!("bridge unlock must be sent from a bridge account"))?;
+
+        // ensure the bridge account's tracked asset is being transferred.
+        let allowed_asset_id = state
+            .get_bridge_account_asset_ids(&from)
```

Comment:
since this returns just 1 asset id now, can you rename the fn to `get_bridge_account_asset_id`?